tests/virt_autotest/validate_host_and_guests.pm

```diff
@@ -0,0 +1,48 @@
+# SUSE's openQA tests
+#
+# Copyright 2012-2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Do a basic examination to the host and guests after tests run
+# Maintainer: qe-virt@suse.de
+
+package validate_host_and_guests;
+
+use strict;
+use warnings;
+use base "virt_feature_test_base";
+use testapi;
+use virt_autotest::common;
+use virt_autotest::utils qw(check_host_health check_guest_health);
+
+sub run_test {
+
+    my $failures = check_host_health;
+    die "Host is not healthy: $failures" unless $failures eq "";
+    foreach my $guest (keys %virt_autotest::common::guests) {
+        $failures = check_guest_health($guest);
+        die "Guest $guest is not healthy: $failures" unless $failures eq "";
+    }
+}
+
+sub prepare_run_test {
+    script_run("rm -f /root/{commands_history,commands_failure}");
+    assert_script_run("history -c");
+}
+
+sub post_fail_hook {
+    my $self = shift;
+    collect_virt_system_logs();
+    virt_utils::collect_host_and_guest_logs;
+    upload_logs("/var/log/clean_up_virt_logs.log");
+    upload_logs("/var/log/guest_console_monitor.log");
+    script_run("rm -f -r /var/log/clean_up_virt_logs.log /var/log/guest_console_monitor.log");
+    $self->upload_coredumps;
+    save_screenshot;
+}
```

Comment:
I do no think this module needs post_fail_hook anymore. It further expands testing scope to nowhere. You already checked what you want, namely "healthy problem", then just leave them alone after such checking. Keep it simple, otherwise this looks like another "test module" which does not serve your original purpose anymore. IMHO.

Reply:
It is another test module.
As we talked yesterday, only passed test will go to this 'validate' test step. Logs will not be uploaded for a passed test. So the post_fail_hook here has to upload logs since the validation failed in this module.
It cannot use the post_fail_hook in base class either, because a health check is done there. So it need its own post_fail_hook.

Reply:
Keep it simple, straightforward and informative.

Reply:
This module should be 
```
fatal => 0
```

Reply:
> This module should be
> 
> ```
> fatal => 0
> ```

No, Alice requests this validation to mark the test fail on purpose as there are bugs, after all our tests have completed.

Reply:
If you are going to make it as fatal, you'd better do not use "Call Trace" as per my [this comment](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/15778/files#r1013574348). We should make sure our test reasonable and suitable.